util/src/keys/store.rs

```diff
@@ -408,7 +412,7 @@ impl EncryptedHashMap<H128> for SecretStore {
 	}
 
 	fn insert<Value: FromRawBytesVariable + BytesConvertable>(&mut self, key: H128, value: Value, password: &str) -> Option<Value> {
-		let previous = if let Ok(previous_value) = self.get(&key, password) { Some(previous_value) } else { None };
+		let previous = if self.exists(&key) { Some(self.get(&key, password).expect("we just checked it existsed")) } else { None };
```

Comment:
typo,
also this is a subject to a race condition. If another process has deleted the key I'd prefer not to panic here